src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java

```diff
@@ -114,18 +114,54 @@ concurrent requests (controlled by --remote_max_concurrency_per_connection), so
       help = "A path to a directory where the corrupted outputs will be captured to.")
   public PathFragment remoteCaptureCorruptedOutputs;
 
+  /**
+   * Determines the mode for uploading action results to a remote or disk cache.
+   *
+   * <p>This enum is used with {@link BoolOrEnumConverter} to support backwards compatibility with
+   * the old boolean {@code --remote_cache_async} flag.
+   */
+  public enum RemoteCacheAsync {
+    /**
+     * Synchronous uploads: block completion of each action until its outputs are uploaded. This is
+     * the old {@code --remote_cache_async=false} behavior.
+     */
+    FALSE,
```

Comment:
We can use more descriptive names for the enum values, how about (feel free to suggest other names) `OFF`, `WAIT_AT_END_OF_BUILD` and `WAIT_AT_NEXT_COMMAND`?

Reply:
I've renamed these to be more descriptive and somewhat aligned with their equivalents for the BES flag.